fs/inode/fs_files.c

```diff
@@ -432,7 +432,7 @@ int nx_dup2(int fd1, int fd2)
                                  [fd2 % CONFIG_NFILE_DESCRIPTORS_PER_BLOCK]);
   _files_semgive(list);
 
-  return ret < 0 ? ret : fd2;
+  return ret;
```

Comment:
The return of `nx_dup2` is used directly in `dup2` and, as mentioned in https://github.com/apache/incubator-nuttx-apps/pull/708, it should be the new file descriptor.

Reply:
Yes, you are right.